cmd/internal/database/contract.go

```diff
@@ -3,6 +3,7 @@ package database
 type DatabaseInitializer interface {
 	Check() (bool, error)
 	Recover() error
+	Upgrade() error
```

Comment:
We probably need a `CheckUpgrade() (bool, error)`, similar to `Check()` in order to figure out if an update can be executed or not. If the update is going to be run, we need to set the API status from the initializer like:

```golang
	i.log.Info("upgrading database")
	i.currentStatus.Status = v1.StatusResponse_UPGRADING
	i.currentStatus.Message = "upgrading database"
```

Reply:
This is not worth the effort.